src/runtime/pipeline/pipeline_struct.h

```diff
@@ -0,0 +1,152 @@
+   * \param reader Json reader.
+   */
+  void Load(dmlc::JSONReader* reader) {
+    reader->BeginArray();
+    while (reader->NextArrayItem()) {
+      std::string key;
+      reader->BeginObject();
+      std::string inputName;
+      int mod_idx;
+      while (reader->NextObjectItem(&key)) {
+        if (key == "mod_idx") {
+          reader->Read(&mod_idx);
+        }
+        if (key == "input_name") {
+          reader->Read(&inputName);
+        }
+      }
+      bindings[mod_idx] = inputName;
+    }
+  }
+};
+/*!
+ * \brief Binding information of the outputs by each module.
+ */
+struct OutputMap {
+  /*! \brief output and output binding map. */
+  std::unordered_map<int, OutputBindings> output_binding_map;
+  OutputMap& operator=(const OutputMap& output) {
+    output_binding_map = output.output_binding_map;
+    return *this;
+  }
+  /*! \brief Global output is the final outputs of pipeline, this function use to
+   *   get how many global outputs are in this Outputmap
+   *  \return Number of global outputs.
+   */
+  size_t GetGlobalOutputNum(void) const {
+    size_t outputNum = 0;
+    for (auto bindings : output_binding_map) {
+      outputNum += bindings.second.IsGlobalOutputNum() ? 1 : 0;
+    }
+    return outputNum;
+  }
+
+  /*!
+   * \brief Create output and output binding map from JSONReader.
+   * \param reader Json reader.
+   */
+  void Load(dmlc::JSONReader* reader) {
+    reader->BeginArray();
+    while (reader->NextArrayItem()) {
+      std::string key;
+      reader->BeginObject();
+      int output_idx;
+      OutputBindings binding;
+      while (reader->NextObjectItem(&key)) {
+        if (key == "output_idx") {
+          reader->Read(&output_idx);
+        }
+        if (key == "dependent") {
+          reader->Read(&binding);
+        }
+      }
+      output_binding_map[output_idx] = binding;
+    }
+  }
+};
+/*!
+ * \brief Binding or dependency information of each module output interface.
+ */
+struct PipelineConfigure {
+  /*!  */
+  std::unordered_map<int, OutputMap> config;
+  OutputMap& operator[](const int key) { return config[key]; }
+  /*!
+   * \brief Get total global outputs number.
+   * \return Global outputs number.
+   */
+  size_t GetGlobalOutputNum() const {
+    size_t output_num = 0;
+    for (auto mod_output : config) {
+      output_num += mod_output.second.GetGlobalOutputNum();
+    }
+    return output_num;
+  }
+};
+/*!
+ * \brief Informations used to initialize the graph executor module, these
+    information comming from export library function call.
+ */
+struct ModuleInformation {
+  std::unordered_map<std::string, std::string> info;
+  const std::string& operator[](const std::string& key) { return info[key]; }
+  ModuleInformation& operator=(const std::unordered_map<std::string, std::string>& umap) {
+    info = umap;
+    return *this;
+  }
+};
+/*! Module information of each module. */
+typedef std::unordered_map<int, ModuleInformation> ModuleConfigure;
```

Comment:
using ModuleConfig = std::unordered_map<int, ModuleInformation>

Document what is int

Reply:
fixed.